2008-06-13  Ulrich Drepper  <drepper@redhat.com>

        * sysdeps/posix/getaddrinfo.c: Move _res_hconf_init call to a
        better place so it is not called when nscd is used.

2008-05-14  Ulrich Drepper  <drepper@redhat.com>

        * sysdeps/posix/getaddrinfo.c (getaddrinfo): Call _res_hconf_init
        if necessary.
        * posix/tst-rfc3484.c: Add dummy definition of _res_hconf_init.
        * posix/tst-rfc3484-2.c: Likewise.

Index: b/posix/tst-rfc3484-2.c
===================================================================
--- a/posix/tst-rfc3484-2.c	2008-01-10 21:00:37.000000000 +0100
+++ b/posix/tst-rfc3484-2.c	2008-08-08 18:38:04.000000000 +0200
@@ -35,6 +35,11 @@
   *output = NULL;
   return 0;
 }
+void
+attribute_hidden
+_res_hconf_init (void)
+{
+}
 
 #include "../sysdeps/posix/getaddrinfo.c"
 
Index: b/posix/tst-rfc3484.c
===================================================================
--- a/posix/tst-rfc3484.c	2008-01-10 21:00:37.000000000 +0100
+++ b/posix/tst-rfc3484.c	2008-08-08 18:38:04.000000000 +0200
@@ -35,6 +35,11 @@
   *output = NULL;
   return 0;
 }
+void
+attribute_hidden
+_res_hconf_init (void)
+{
+}
 
 #include "../sysdeps/posix/getaddrinfo.c"
 
Index: b/sysdeps/posix/getaddrinfo.c
===================================================================
--- a/sysdeps/posix/getaddrinfo.c	2008-08-08 18:37:55.000000000 +0200
+++ b/sysdeps/posix/getaddrinfo.c	2008-08-08 18:38:04.000000000 +0200
@@ -60,6 +60,7 @@
 #include <not-cancel.h>
 #include <nscd/nscd-client.h>
 #include <nscd/nscd_proto.h>
+#include <resolv/res_hconf.h>
 
 #ifdef HAVE_LIBIDN
 extern int __idna_to_ascii_lz (const char *input, char **output, int flags);
@@ -670,6 +671,9 @@
 					     "dns [!UNAVAIL=return] files",
 					     &nip);
 
+	  /* Initialize configurations.  */
+	  if (__builtin_expect (!_res_hconf.initialized, 0))
+	    _res_hconf_init ();
 	  if (__res_maybe_init (&_res, 0) == -1)
 	    no_more = 1;
 
